sisyphus/graph.py

```diff
@@ -489,69 +489,77 @@ def for_all_nodes(self, f, nodes=None, bottom_up=False):
                     if path.creator:
                         nodes.append(path.creator)
 
-        visited = {}
-        finished = 0
-
         if gs.GRAPH_WORKER == 1:
-            # Run in main thread if only one graph worker is given anyway
-            def runner(job):
-                """
-                :param Job job:
-                """
-                # make sure all inputs are updated
+            visited_set = set()
+            visited_list = []
+            queue = list(reversed(nodes))
```

Comment:
Because I pop the latest entry (`queue.pop(-1)`) and I wanted to keep a similar order as before (although it might not matter; the order of `job._sis_inputs` is anyway random).